Editor/Window/UserProfileSelection.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Linq;
+using System.Threading;
+using Editor.CoreAPI;
+using UnityEngine.UIElements;
+
+namespace AmazonGameLift.Editor
+{
+    internal class UserProfileSelection
+    {
+        private BootstrapSettings _bootstrapSettings;
+        private CancellationTokenSource _refreshBucketsCancellation;
+        private readonly VisualElement _container;
+        private readonly AwsUserProfilesPage _awsUserProfilesPage;
+        private readonly StateManager _stateManager;
+        
+        public UserProfileSelection(VisualElement container, StateManager stateManager, AwsUserProfilesPage profilesPage)
+        {
+            _container = container;
+            _stateManager = stateManager;
+            _awsUserProfilesPage = profilesPage;
+            _bootstrapSettings = profilesPage.BootstrapSettings;
+        }
+
+        private void BucketSelection()
+        {
+            _refreshBucketsCancellation = new CancellationTokenSource();
+            _ = _bootstrapSettings.RefreshExistingBuckets(_refreshBucketsCancellation.Token);
+            if (_bootstrapSettings.BucketName != null)
+            {
+                BucketSelection(_bootstrapSettings.BucketName);
+                _bootstrapSettings.SaveSelectedBucket();
+            }
```

Comment:
RefereshExistingBuckets doesn't set the BucketName property.  So if you don't already have it, this isn't going to be set. 

We should just have a single method `SelectBucket(string selectedBucket)` and the invoker decides what to do based on whether a bucketName is already available from `_bootstrapSettings.BucketName`. Please let me know if that makes sense.

Reply:
UserProfileSelection is no longer used, removed;